### PR TITLE
[FIX] Corpus: remove text features which not in metas

### LINKED ISSUE #324: Copying corpus sometimes fails
##### Text version
Current (~**0.2.5**) and older

##### Orange version
master

##### Expected behavior
```python
corpus = Corpus(...)
new_corpus = corpus.copy()
```
That should work and not raise an error.

##### Actual behavior

> orangecontrib.text.corpus in set_text_features at line 90
>    raise ValueError('Feature "{}" not found.'.format(f))
> orangecontrib.text.corpus in __init__ at line 73
>    self.set_text_features(text_features)
> orangecontrib.text.corpus in copy at line 321
>    self.W.copy(), copy(self.text_features))
> orangecontrib.text.widgets.owpreprocess in set_data at line 598
>    self.corpus = data.copy() if data is not None else None

https://sentry.io/biolab/orange3-text-bl/issues/399242697/

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
Fixes #324 

##### Includes
- [X] Code changes
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: orangecontrib/text/corpus.py [hunk 1]
```diff
@@ -434,8 +434,7 @@ def retain_preprocessing(orig, new, key=...):
                 else:
                     raise TypeError('Indexing by type {} not supported.'.format(type(key)))
                 new._dictionary = orig._dictionary
-
-            new.text_features = orig.text_features
+            new.text_features = list(set(orig.text_features) & set(new.domain.metas)) or None
```
Comment (nikicc) [inline orangecontrib/text/corpus.py]: Why or `None`? Perhaps it's safer to set `text_features` to an empty list.

File: orangecontrib/text/tests/test_corpus.py [hunk 2]
```diff
@@ -389,3 +389,15 @@ def test_from_documents(self):
...
+        """
+        Remove those text features which do not have a column in metas.
+        GH-324
+        GH-325
+        """
+        c = Corpus.from_file('friends-transcripts')
+        domain = Domain(attributes=c.domain.attributes, class_vars=c.domain.class_vars)
+        d = c.transform(domain)
+        self.assertTrue(d.text_features is None or d.text_features[0] in d.domain.metas)
+        self.assertTrue(d.text_features is None or d.text_features[0] in d.domain.metas)
```
Comment (nikicc) [inline orangecontrib/text/tests/test_corpus.py]: Isn't this the same as above line?

File: orangecontrib/text/tests/test_corpus.py [hunk 3]
```diff
@@ -389,3 +389,15 @@ def test_from_documents(self):
...
+    def test_corpus_remove_text_features(self):
+        """
+        Remove those text features which do not have a column in metas.
+        GH-324
+        GH-325
+        """
+        c = Corpus.from_file('friends-transcripts')
+        domain = Domain(attributes=c.domain.attributes, class_vars=c.domain.class_vars)
+        d = c.transform(domain)
+        self.assertTrue(d.text_features is None or d.text_features[0] in d.domain.metas)
```
Comment (nikicc) [inline orangecontrib/text/tests/test_corpus.py]: Could we make thi check without an `or`.

File: orangecontrib/text/tests/test_corpus.py [hunk 4]
```diff
@@ -389,3 +389,15 @@ def test_from_documents(self):
...
+        self.assertTrue(d.text_features is None or d.text_features[0] in d.domain.metas)
+        d.copy()
```
Comment (nikicc) [inline orangecontrib/text/tests/test_corpus.py]: Should we add a comment here?

File: orangecontrib/text/corpus.py (hunk 1 above)
Comment (nikicc) [reply orangecontrib/text/corpus.py]: Perhaps also fix this in `__init__`?

File: orangecontrib/text/tests/test_corpus.py [hunk 5]
```diff
@@ -389,3 +389,15 @@ def test_from_documents(self):
         self.assertEqual([engine_dv.repr_val(v) for v in c.X[:, 0]],
                          [d['engine'] for d in documents])
 
+    def test_corpus_remove_text_features(self):
+        """
+        Remove those text features which do not have a column in metas.
+        GH-324
+        GH-325
+        """
+        c = Corpus.from_file('friends-transcripts')
```
Comment (nikicc) [inline orangecontrib/text/tests/test_corpus.py]: Perhaps, use `deerwester` that loads much faster?

File: orangecontrib/text/corpus.py (hunk 1 above)
Comment (jerneju) [reply orangecontrib/text/corpus.py]: `self.text_features = None` changed to `self.text_features = []`
`text_features=None` as a default argument in `__init__` is kept.

**Pull request merged**